### PR TITLE
Pal 391 policy service cachable

### PR DESCRIPTION
**Pull request by dev930018**

### CODE REVIEW

File: policy-service-integration/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,24 @@
+# Copyright 2019 Crown Copyright
+#
```
Comment (pd104923) [inline policy-service-integration/Dockerfile]: Update copyright year

File: policy-service-integration/pom.xml [hunk 2]
```diff
@@ -23,8 +23,192 @@
         <groupId>uk.gov.gchq.palisade</groupId>
```
Comment (pd104923) [inline policy-service-integration/pom.xml]: The copyright needs to be updated

File: policy-service-integration/src/test/resources/application.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
+# Copyright 2019 Crown Copyright
+#
```
Comment (pd104923) [inline policy-service-integration/src/test/resources/application.yaml]: Update copyright

File: policy-service-integration/src/test/resources/bootstrap.yaml [hunk 4]
```diff
@@ -0,0 +1,20 @@
+# Copyright 2019 Crown Copyright
+#
```
Comment (pd104923) [inline policy-service-integration/src/test/resources/bootstrap.yaml]: Update copyright

File: charts/policy-service-integration/values.yaml [hunk 5]
```diff
@@ -0,0 +1,35 @@
+# Copyright 2019 Crown Copyright
```
Comment (developer01189998819991197253) [inline charts/policy-service-integration/values.yaml]:
```suggestion
# Copyright 2020 Crown Copyright
```

File: policy-service-integration/pom.xml [hunk 6]
```diff
@@ -23,8 +23,192 @@
...
+                <includes>
+                    <include>**/values.yaml</include>
+                </includes>
+            </resource>
+            <resource>
+                <directory>src/main/resources</directory>
+                <filtering>false</filtering>
+                <excludes>
+                    <exclude>**/values.yaml</exclude>
+                </excludes>
```
Comment (ac74475) [inline policy-service-integration/pom.xml]: these two seem contradictory

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    @Test
+    public void isUp() {
+        Response health = policyClient.getActuatorHealth();
+
+        assertThat(health.status(), equalTo(200));
+    }
+
+    @Ignore // TODO: This test is failing with a 400 error (cannot use PassThroughRule as a subclass of Rule)
```
Comment (ac74475) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java]: Can you fix this issue or added it as a bug in the backlog if it isn't already

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyCachingProxyTest.java [hunk 8]
```diff
@@ -0,0 +1,143 @@
...
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assume.assumeTrue;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest(classes = PolicyApplication.class, webEnvironment = WebEnvironment.NONE)
+public class PolicyCachingProxyTest extends PolicyTestCommon {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PolicyCachingProxyTest.class);
```
Comment (developer01189998819991197253) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyCachingProxyTest.java]:
```suggestion
```
Unused logger

File: policy-service-integration/src/test/resources/application.yaml [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+    allow-bean-definition-overriding: true
+
+server:
+  port: 18085
+
+eureka:
+  instance:
+    hostname: localhost
+    port: 8083
+    preferIpAddress: true
```
Comment (developer01189998819991197253) [inline policy-service-integration/src/test/resources/application.yaml]: pal-567 raised to swap preferIpAddress to false across all repos

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyTestCommon.java [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+
+    static final FileResource newFile = new FileResource().id("/new/file.exe").serialisedFormat("exe").type("elf").parent(new SystemResource().id("/new"));
+
+    // A do-nothing policy to apply to leaf resources
+    static final Policy passThroughPolicy = new Policy<>()
+            .owner(user)
+            .resourceLevelRule("Does nothing", new PassThroughRule<>())
+            .recordLevelRule("Does nothing", new PassThroughRule<>());
+
+    static final Set<DirectoryResource> directoryResources = new HashSet<>(Arrays.asList(jsonDirectory, secretDirectory));
```
Comment (developer01189998819991197253) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyTestCommon.java]: directoryResources is never used. needed?

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java (hunk 7 above)
Comment (dev930018) [reply policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java]: fixed

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyTestCommon.java (hunk 10 above)
Comment (dev930018) [reply policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyTestCommon.java]: not needed at the moment, kept in for completeness

File: policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java [hunk 11]
```diff
@@ -0,0 +1,103 @@
...
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.equalTo;
+import static org.junit.Assert.assertNotEquals;
+import static org.junit.Assert.assertNotNull;
+
+@RunWith(SpringRunner.class)
+@EnableFeignClients
+@SpringBootTest(classes = PolicyApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+public class PolicyComponentTest extends PolicyTestCommon {
+    private static final Logger LOGGER = LoggerFactory.getLogger(PolicyCachingProxyTest.class);
```
Comment (developer01189998819991197253) [inline policy-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/policy/PolicyComponentTest.java]:
```suggestion
    private static final Logger LOGGER = LoggerFactory.getLogger(PolicyComponentTest.class);
```

**Pull request merged**